--- libtool-2.4.6/build-aux/ltmain.in.0012~	2015-07-12 16:08:21.127560720 +0200
+++ libtool-2.4.6/build-aux/ltmain.in	2015-07-12 16:10:25.849263536 +0200
@@ -5657,7 +5657,10 @@ func_mode_link ()
 	case $pass in
 	dlopen) libs=$dlfiles ;;
 	dlpreopen) libs=$dlprefiles ;;
-	link) libs="$deplibs %DEPLIBS% $dependency_libs" ;;
+	link)
+          libs="$deplibs %DEPLIBS%"
+          test "X$link_all_deplibs" != Xno && libs="$libs $dependency_libs"
+          ;;
 	esac
       fi
       if test lib,dlpreopen = "$linkmode,$pass"; then
--- libtool-2.4.6/m4/libtool.m4.0012~	2015-07-12 16:08:21.121560787 +0200
+++ libtool-2.4.6/m4/libtool.m4	2015-07-12 16:12:24.857153361 +0200
@@ -4939,6 +4939,9 @@ m4_if([$1], [CXX], [
       ;;
     esac
     ;;
+  linux* | k*bsd*-gnu)
+    _LT_TAGVAR(link_all_deplibs, $1)=no
+    ;;
   *)
     _LT_TAGVAR(export_symbols_cmds, $1)='$NM $libobjs $convenience | $global_symbol_pipe | $SED '\''s/.* //'\'' | sort | uniq > $export_symbols'
     ;;
@@ -5468,6 +5471,7 @@ _LT_EOF
 	# /with/ Import File - we do not want to mix them.
 	shared_flag_aix='-shared'
 	shared_flag_svr4='-shared $wl-G'
+        _LT_TAGVAR(link_all_deplibs, $1)=no
       else
 	# not using gcc
 	if test ia64 = "$host_cpu"; then
@@ -6538,6 +6542,7 @@ if test yes != "$_lt_caught_CXX_error";
 	  # /with/ Import File - we do not want to mix them.
 	  shared_flag_aix='-shared'
 	  shared_flag_svr4='-shared $wl-G'
+          _LT_TAGVAR(link_all_deplibs, $1)=no
         else
           # not using gcc
           if test ia64 = "$host_cpu"; then
