--- libtool-2.4.6/build-aux/ltmain.in.0013~	2015-07-12 16:13:04.092810000 +0200
+++ libtool-2.4.6/build-aux/ltmain.in	2015-07-12 16:14:18.645184139 +0200
@@ -5527,6 +5527,12 @@ func_mode_link ()
       fi
     done # argument parsing loop
 
+    if test "$module" = yes ; then
+      # [OpenMandriva] override ld option "--no-undefined" which is wrong for plugins
+      linker_flags="$linker_flags --warn-unresolved-symbols"
+      compiler_flags="$compiler_flags -Wl,--warn-unresolved-symbols"
+    fi
+
     test -n "$prev" && \
       func_fatal_help "the '$prevarg' option requires an argument"
 
