--- libtool-2.4.6/m4/ltdl.m4.0020~	2015-07-12 16:21:29.308057047 +0200
+++ libtool-2.4.6/m4/ltdl.m4	2015-07-12 16:22:28.481672449 +0200
@@ -95,7 +95,7 @@ m4_defun([_LTDL_CONVENIENCE],
   "") enable_ltdl_convenience=yes
       ac_configure_args="$ac_configure_args --enable-ltdl-convenience" ;;
 esac
-LIBLTDL='_LT_BUILD_PREFIX'"${lt_ltdl_dir+$lt_ltdl_dir/}libltdlc.la"
+LIBLTDL='_LT_BUILD_PREFIX'"${lt_ltdl_dir+$lt_ltdl_dir/}libltdlc.so"
 LTDLDEPS=$LIBLTDL
 LTDLINCL='-I$(top_srcdir)'"${lt_ltdl_dir+/$lt_ltdl_dir}"
 
@@ -166,7 +166,7 @@ case $enable_ltdl_install in
       ;;
   *)  enable_ltdl_install=yes
       ac_configure_args="$ac_configure_args --enable-ltdl-install"
-      LIBLTDL='_LT_BUILD_PREFIX'"${lt_ltdl_dir+$lt_ltdl_dir/}libltdl.la"
+      LIBLTDL='_LT_BUILD_PREFIX'"${lt_ltdl_dir+$lt_ltdl_dir/}libltdl.so"
       LTDLDEPS=$LIBLTDL
       LTDLINCL='-I$(top_srcdir)'"${lt_ltdl_dir+/$lt_ltdl_dir}"
       ;;
@@ -273,10 +273,10 @@ fi
 
 AC_ARG_WITH([ltdl_lib],
     [AS_HELP_STRING([--with-ltdl-lib=DIR],
-                    [use the libltdl.la installed in DIR])])
+                    [use the libltdl.so installed in DIR])])
 
 if test -n "$with_ltdl_lib"; then
-  if test -f "$with_ltdl_lib/libltdl.la"; then :
+  if test -f "$with_ltdl_lib/libltdl.so"; then :
   else
     AC_MSG_ERROR([invalid ltdl library directory: '$with_ltdl_lib'])
   fi
