diff --git a/tests/configure-iface.at b/tests/configure-iface.at
index 47e56f0..724c539 100644
--- a/tests/configure-iface.at
+++ b/tests/configure-iface.at
@@ -140,7 +140,7 @@ main_DEPENDENCIES	= $(LTDLDEPS) libmodule.la
 prefix=`pwd`/_inst
 
 LT_AT_BOOTSTRAP([--ltdl], [-I libltdl/m4], [], [--add-missing], [],
-        [--with-included-ltdl --prefix=$prefix], [all install])
+        [--with-included-ltdl --prefix=$prefix --enable-ltdl-install], [all install])
 
 # TODO: Check that the installed program 'main' is linked against our libltdl
 AT_CHECK([test -f "$prefix/lib/libltdl.la"])
