diff -up libtool-2.4.7/build-aux/ltmain.in.12~ libtool-2.4.7/build-aux/ltmain.in
--- libtool-2.4.7/build-aux/ltmain.in.12~	2022-09-08 16:46:01.547437354 +0200
+++ libtool-2.4.7/build-aux/ltmain.in	2022-09-08 16:46:38.722900438 +0200
@@ -5409,10 +5409,11 @@ func_mode_link ()
       # -fsanitize=*         Clang/GCC memory and address sanitizer
       # -fuse-ld=*           Linker select flags for GCC
       # -Wa,*                Pass flags directly to the assembler
+      # --rtlib=*            clang runtime library (libgcc vs compiler-rt)
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*|--sysroot=*| \
       -O*|-g*|-flto*|-fwhopr*|-fuse-linker-plugin|-fstack-protector*|-stdlib=*| \
-      -specs=*|-fsanitize=*|-fuse-ld=*|-Wa,*)
+      -specs=*|-fsanitize=*|-fuse-ld=*|-Wa,*|--rtlib=*)
         func_quote_arg pretty "$arg"
 	arg=$func_quote_arg_result
         func_append compile_command " $arg"
