diff -up libtool-2.4.7/build-aux/ltmain.in.12~ libtool-2.4.7/build-aux/ltmain.in
--- libtool-2.4.7/build-aux/ltmain.in.12~	2022-09-08 16:48:23.002527238 +0200
+++ libtool-2.4.7/build-aux/ltmain.in	2022-09-08 16:52:06.285708857 +0200
@@ -5410,10 +5410,18 @@ func_mode_link ()
       # -fuse-ld=*           Linker select flags for GCC
       # -Wa,*                Pass flags directly to the assembler
       # --rtlib=*            clang runtime library (libgcc vs compiler-rt)
+      # -f*                  Mostly optimization related flags that become
+      #                      relevant at link time when LTO is enabled -- also
+      #                      bits that are relevant for picking runtime libraries
+      #                      and tools (-fopenmp, -fuse-ld, ...)
+      # -polly*              Clang polly optimizer flags, may be used with LTO
+      # -pipe                GCC/Clang launch ld through a pipe instead of temp files
+      # --param=*            GCC/Clang parameters (ssp-buffer-size=4, ...)
+      # -Wl,*                GCC/Clang obviously meant for the linker
       -64|-mips[0-9]|-r[0-9][0-9]*|-xarch=*|-xtarget=*|+DA*|+DD*|-q*|-m*| \
       -t[45]*|-txscale*|-p|-pg|--coverage|-fprofile-*|-F*|@*|-tp=*|--sysroot=*| \
       -O*|-g*|-flto*|-fwhopr*|-fuse-linker-plugin|-fstack-protector*|-stdlib=*| \
-      -specs=*|-fsanitize=*|-fuse-ld=*|-Wa,*|--rtlib=*)
+      -specs=*|-fsanitize=*|-fuse-ld=*|-Wa,*|--rtlib=*|-f*|-polly*|-pipe|--param=*|-Wl,*)
         func_quote_arg pretty "$arg"
 	arg=$func_quote_arg_result
         func_append compile_command " $arg"
