--- libtool-2.4.6/tests/configure-iface.at.0017~	2015-07-12 16:19:14.758964288 +0200
+++ libtool-2.4.6/tests/configure-iface.at	2015-07-12 16:19:56.049680538 +0200
@@ -142,7 +142,7 @@ main_DEPENDENCIES	= $(LTDLDEPS) libmodul
 prefix=`pwd`/_inst
 
 LT_AT_BOOTSTRAP([--ltdl], [-I libltdl/m4], [], [--add-missing], [],
-        [--with-included-ltdl --prefix=$prefix], [all install])
+        [--with-included-ltdl --prefix=$prefix --enable-ltdl-install], [all install])
 
 # TODO: Check that the installed program 'main' is linked against our libltdl
 AT_CHECK([test -f "$prefix/lib/libltdl.la"])
