--- libtool-2.4.6/build-aux/ltmain.in.0000~	2015-07-12 15:55:44.866613700 +0200
+++ libtool-2.4.6/build-aux/ltmain.in	2015-07-12 15:58:02.767136151 +0200
@@ -7818,15 +7818,23 @@ EOF
 	fi
 
 	tmp_deplibs=
+        inst_prefix_arg=
 	for test_deplib in $deplibs; do
 	  case " $convenience " in
 	  *" $test_deplib "*) ;;
 	  *)
-	    func_append tmp_deplibs " $test_deplib"
+            if test -n "$inst_prefix_dir" && test "$inst_prefix_dir" != "/" && (echo "$test_deplib" | grep -- "$inst_prefix_dir" >/dev/null); then
+              func_append inst_prefix_arg " $test_deplib"
+            else
+	      func_append tmp_deplibs " $test_deplib"
+            fi
 	    ;;
 	  esac
 	done
 	deplibs=$tmp_deplibs
+        if test -n "$inst_prefix_arg"; then
+          deplibs="$inst_prefix_arg $deplibs"
+        fi
 
 	if test -n "$convenience"; then
 	  if test -n "$whole_archive_flag_spec" &&
